[ButtonBase] Consider as a link with a custom component and `to` prop (#26576)
mui-org/material-ui: docs/src/pages/guides/routing/routing.md

## `component` prop

You can achieve the integration with third-party routing libraries with the `component` prop.
You can learn more about this prop in the [**composition guide**](/guides/composition/#component-prop).

### Link

Here are a few demos with [react-router](https://github.com/ReactTraining/react-router).
You can apply the same strategy with all the components: BottomNavigation, Card, etc.

{{"demo": "pages/guides/routing/LinkRouter.js"}}

### Button

{{"demo": "pages/guides/routing/ButtonRouter.js"}}

**Note**: The button base component adds the `role="button"` attribute when it identifies the intent to render a button without a native `<button>` element.
This can create issues when rendering a link.

> If you are not using one of the `href`, `to`, or `component="a"` props, you need to **swallow the role** attribute.
> It's why `role={undefined}` is set in the above demo.

Reviewer: ```suggestion
If you are not using one of the `href`, `to`, or `component="a"` props, you need to override the `role` attribute.
The above demo achieves this by setting `role={undefined}` **after** the spread props.
```

Author: @oliviertassinari should I do these changes?

Here are a few demos with [react-router](https://github.com/ReactTraining/react-router).
You can apply the same strategy with all the components: BottomNavigation, Card, etc.

> **Note**: When the `component` prop is used on the `ButtonBase` and not indicators are

Reviewer: ```suggestion
**Note**: When the `component` prop is used on the `ButtonBase` and no indicators are
```

Author: fixed

> **Note**: When the `component` prop is used on the `ButtonBase` and not indicators are

Reviewer: What are these "indicators"?

Author: Maybe I can change it like this?
 

 **Note**: When the `component` prop is used on the `ButtonBase` and no value for the prop `role` is provided then, `role="button"` is provided by the `ButtonBase` to the component by default.
You need to override this attribute if you are rendering a link, e.g. with `role={undefined}`.

Reviewer 2: The indicator would be the href or to props.

Author: @eps1lon is it fine now?
I have updated the changes you asked.

given that a link is rendered, a `role="button"` prop is provided by the `ButtonBase` to the component.

> You need to swallow this attribute if you are rendering a link, e.g. with `role={undefined}`.

Reviewer: ```suggestion
You need to override this attribute if you are rendering a link, e.g. with `role={undefined}`.
```

Author: fixed

> For instance:

Author: @oliviertassinari just want to get this to your notice a small miss spell is there `componnet="a"`
